lib/broccoli/angular2-app.js

```diff
@@ -41,33 +41,31 @@ Angular2App.prototype.toTree = function() {
       'angular2/bundles/upgrade.dev.js'
     ];
 
-
-
     if (this.options && this.options.vendorNpmFiles) {
       vendorNpmFiles = vendorNpmFiles.concat(this.options.vendorNpmFiles);
     }
 
     var tsconfig = JSON.parse(fs.readFileSync('src/tsconfig.json', 'utf-8'));
-    var tsConfigCompilerOptions = tsconfig.compilerOptions;
-
-
-    // TODO(i): why do we need these additional paths? remove?
-    tsConfigCompilerOptions.rootFilePaths = this.additionalPaths.map(function(name) {
-        return path.join(process.cwd(), sourceDir, name)
-      });
-
-    // TODO(i): kill rootFilePaths in broccoli-typescript and use tsconfig.json#files instead
-    tsConfigCompilerOptions.rootFilePaths = tsConfigCompilerOptions.rootFilePaths.
-        concat(tsconfig.files.map(function(p) {
-          // TODO(i): this is a hack - for some reason we need to prefix all paths with srcDir because
-          //          tsc's "rootDir" doesn't take effect when resolving "files" paths
-          return path.join(sourceDir, p);
-        }));
+    // Add all spec files to files. We need this because spec files are their own entry
+    // point.
+    fs.readdirSync(sourceDir).forEach(function addPathRecursive(name) {
+      const filePath = path.join(sourceDir, name);
+      if (filePath.match(/\.spec\.[jt]s$/)) {
+        tsconfig.files.push(name);
+      } else if (fs.statSync(filePath).isDirectory()) {
+        // Recursively call this function with the full sub-path.
+        fs.readdirSync(filePath).forEach(function(n) {
+          addPathRecursive(path.join(name, n));
+        });
+      }
+    });
 
+    // Because the tsconfig does not include the source directory, add this as the first path
+    // element.
+    tsconfig.files = tsconfig.files.map(name => path.join(sourceDir, name));
```

Comment:
In line 52, this is already being done for test files. Won't those test files end up having `sourceDir` twice at the start?